### PR TITLE
Jest: Migrate Tooltip.spec.tsx

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
This PR does not strictly fulfil the requirements of the issue. 

The existing tests directly modify the state, which is an antipattern, since effectively implementation details are tested. Thus I do not intend to replicate this test.

Instead I wrote 4 tests which document the behaviour of the Tooltip and added comments to guide for test writing. 

This is an improvement over the existing test and still sufficient for the migration. 

## Related Issue
https://github.com/recharts/recharts/issues/3215

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test-jest/component/Tooltip.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    const mouseOverEvent = mockMouseEvent('mouseover', chart!, { pageX: 200, pageY: 200 });
+    mouseOverEvent.fire();
+
+    // After the mouse over event over the chart, the tooltip wrapper still is not set to visible,
+    // but the content is already created based on the nearest data point.
+    const tooltipContentName = container.querySelector('.recharts-tooltip-item-name');
+    const tooltipContentValue = container.querySelector('.recharts-tooltip-item-value');
+    expect(tooltipContentName).toBeInTheDocument();
+    expect(tooltipContentValue).toBeInTheDocument();
+  });
```
Comment (nikolasrieble) [inline test-jest/component/Tooltip.spec.tsx:67]: If you have any idea / suggestion on how to trigger visibility, I would love to include that in the PR. @ckifer @proke03

Comment (ckifer) [reply test-jest/component/Tooltip.spec.tsx:67]: I think we may need to refactor the Tooltip to use React Portals (theres a FR for that somewhere) in order for this to be easier to test.

I'm okay with merging this and coming back to it

**Review by ckifer — APPROVED** [review]
LGTM, maybe @nikolasrieble we can create an issue to re-visit these tests so we don't forget

File: test-jest/component/Tooltip.spec.tsx (hunk 1 above)
Comment (proke03) [reply test-jest/component/Tooltip.spec.tsx:67]: For now, I think this is the best way.
I tried implementing that code as a userEvent, but it failed.

**Pull request merged**